runtime/metricsview/executor_validate.go

```diff
@@ -267,6 +270,90 @@ func (e *Executor) validateIndividualDimensionsAndMeasures(ctx context.Context,
 	slices.SortFunc(res.MeasureErrs, func(a, b IndexErr) int { return a.Idx - b.Idx })
 }
 
+func (e *Executor) validateAnnotations(ctx context.Context, mv *runtimev1.MetricsViewSpec, res *ValidateMetricsViewResult) {
+	allMeasures := make([]string, 0, len(mv.Measures))
+	for _, m := range mv.Measures {
+		allMeasures = append(allMeasures, m.Name)
+	}
+
+	// Get the controller used for getting the annotation's model
+	ct, err := e.rt.Controller(ctx, e.instanceID)
+	if err != nil {
+		res.OtherErrs = append(res.OtherErrs, fmt.Errorf("failed to get controller: %w", err))
+		return
```

Comment:
Better to return this error – it's not really a validation error, more an execution error (most likely a `ctx` cancellation).